### ISSUE TITLE
Running a single test-case of a theory with inferred generic parameters fails

### ISSUE THREAD

**Issue by estrizhok** [xunit/xunit#2964]
xUnit.v3 0.1.1-pre.489

STR:

1. Create & compile a test project from the following snippet:
```
public class TestClass
{
  [Theory]
  [InlineData("data")]
  public void TheoryWithTypeArgument<T>(T data) { }
}
```
2. Run discovery on it to acquire `_TestCaseDiscovered.Serialization` value of the single test-case
3. Try running it using the following command line:
```
<your_test_project.exe> -run <your_serialization_value>
```

Expected result:
1. The test-case is executed & successful

Actual result:
1. The test-case fails with `System.InvalidOperationException : Late bound operations cannot be performed on types or methods for which ContainsGenericParameters is true.`

Example of the output on my machine:
```
C:\Work\dotnet-libs\ReSharperTestRunner\src\.tests\TestRunner.Adapters.XUnit3.Tests.Container\bin\Debug\net8.0>TestRunner.Adapters.XUnit3.Tests.Container.exe -run -4:WHVuaXQudjMuWHVuaXRUZXN0Q2FzZSx4dW5pdC52My5jb3Jl:ZG46MDpWR2hsYjNKNVYybDBhRVJwYzNCc1lYbE9ZVzFsUVc1a1ZIbHdaVUZ5WjNWdFpXNTBQRk4wY21sdVp6NG9aR0YwWVRvZ0ltUmhkR0VpS1E9PQp0bTotNDpXSFZ1YVhRdWRqTXVWR1Z6ZEUxbGRHaHZaQ3g0ZFc1cGRDNTJNeTVqYjNKbDpiVzQ2TURwV1IyaHNZak5LTlZZeWJEQmhSVkp3WXpOQ2MxbFliRTlaVnpGc1VWYzFhMVpJYkhkYVZVWjVXak5XZEZwWE5UQUtkR002TFRRNlYwaFdkV0ZZVVhWa2FrMTFWa2RXZW1SRlRuTlpXRTU2VEVob01XSnRiREJNYmxsNlRHMU9kbU50VlQwNlpFZE5Oa3hVVVRaV01HaFhaRmRHV1ZWWVZtdGhhekV4Vm10a1YyVnRVa1pVYmxwcFVqTm9jMWRVVGxOalIwbDVUa2hPYkZOR1dqRlpWbWhTWkZkU2NWUllWbHBOYW13MVYyeEZPVkJVY0dGU2VsRXlWRlZTZDFZeFNYaFhhbHBoVWxVMVJGbFhNVXBsVjFaSlZHMUdWMDFFVWpOWFZscHFUbGRTVm1KRmFGaGliSEJ4V1ZaV1MxWnNaSE5oUlRsT1VteGFNVlp0TlZkaFYwcFhWMnBXVlZJelVraFpWRVp6VjFaR2NWRnRSbGhTV0VFeVZrVmtOR0l4V25SVGJsSnBVa1ZLVDFwV1VsZFdiR1J6WVVVNVRsSXdOREZVYkZaVFlWVXhjVlpZWkZoV2JWSjZXa1phZDFkV1RsbFdiR1JUVFZadk1sZHJWbGRVTWsxNFlrWnNWV0p1UWt0VlZFcHZaREZrY2xWdVdtRk5SRll3VmxaU1MxUXlWblJWYms1VlZqTkNTMXBXVlRGU1ZrWjBZa1U1VjFKcldURldSbU40VW0xUmQwMVdWbGRYU0VKb1ZXdFdSMDVzVW5SalJVcHNVbFJHTUZaV1VrdFViRnBIVmxSR1dHRXlVa2haVjNNeFpFWktXV0ZGTlZOU1ZsWTBWakZhVTFack1IZE9WbFpZWWxSR1QxVnFSbkpOUmxKWlkwVndUbUY2UmxaVk1WSkRWREF4YzFkdE9WaGhNbEpUVkd0Vk1WSkhSalpOUlhSclVqQlZNbFJHVWxKT2JGbDNZVVprYTFZd1dscFdWbWhYWVRKR2NrMVVSbGRoTWxKWVdsY3hVMUpzU25WalIzQk9Za1p2ZDFZeFkzaE9SVFZXWlVWc2FGSkZXbkJaYkdRelpERlNTRTVXY0d4aE0yZ3dWa2MxWVdGdFNsZFdWR3hRWWtkNFdsVldVbUZVYkVwSlVXeE9ZVlo2VmtSWlYzaGhZMVpPVlZadGVGZE5SbGt3VjJ4V2EwMUhSbFpOU0doVVlXdGFXVmxyWkc5Vk1YQkdXa1YwVkZJd2NFbFphMlIzVkd4a1NHRkdWbGROYm1ob1ZrUktTbVZIU2tkaVJsSnBZbFpLYjFacVFsZGtNVnB6V2toT2FGSlViRTlWYlhoaFRWWmFkRTVXWkdoV01IQllWakp3VjFkR1NsaGhSa0poVm14d2VWcFZaRkpsYkZwMFpFWk9UbFp0T0hsV01uUlhZakZSZUdFemFGaGhNbWhZV1cwMVEyTnNWblJrU0Zwc1VtMTRNRlJWVWtkaFZrbDNUbGhrVldKR1dreFhWbHBLWkRGa2NWUnRSbE5XTVVwWlYyeGFhMVF3TlZaT1ZtUnBVbTFTV0ZwWGVHRk5NV1J5Vm14T1ZXSldTbGhXTW5SdlZqSktkVkZ1U2xaTlJsVjRXVmQ0YTFkRk1WbFViWFJYWWxkb1NGWnNZM2hPUm1SMFUyNU9XR0pzV2xsVmJHUk9aVVpzY2xwR1RsTldiVGsyVm0xek1WVXlTblJoUm1SWFRXNW9kbFpFUVhoU01WcDFWbXhrYVdKR2NIZFhWM2hoV1ZVMVYxVnJhR3RTVjFKdlZGWmtORmRzVmxoalJVNWFWbTFTUjFrd1VtRldWa2w2WVVoYVYySlVSa3hhUldSTFVsWkdjazVYZUZkU2JIQllWbXBLTkdJeFVYZE5WV2hWWVd4YVVGWnNXa3RqUmxKWVkwWmFhMDFYVWxoWGEyaFBZVEZhZFZGc2NGWk5WMmhZVjFaYVNtVnRSa2xqUlZKcFZsUldNRmRYTVhka2JWRjNWRzVXV0ZaR2NFOVZhMmhEVkRGYVdXTkdUazVOVmtwSldUQldTMkV4V2xWU2EwNVdVbXhHTTFWR1JuZGpSbkJGWWpOa1VHRjZSVEpaYTJSelZERktSbFpxVWxoaVIxSk1XVlpWTlZaV2NFaGtSMFpUVW14V05GWkhNWGRWYXpWSFlrWm9WMVpHV2xCV2ExcFhUbXhTVjFWc1NrNVNiWGN5VmpJeE5GbFhSbk5XV0dSWVZrVndVRmxYY3pWU1ZrcFZWbXMxYkdKR1ZqVldSbEpMVm1zMVJrNVlVbFJXUmxwaFZtcEdSazFXVW5OVmJVWm9WbFJzVmxWV1VrOVVNa1p5VGxoR1ZWWjZSa2RhVjNNeFlteENWVTFFTUV0Wk1rWjFUMnBCTmxaclpGZGxiVkpIVTJwR2FXSlVWbk5aTW1zeFVXeHdTRkp1Wkd0U01WbzFXVE5yTVZkV1dsaE9XRUpyVWtVeE1WWnJaRmRsYlZKSlZGaFdVazFxYkRGYVJXUkhZMGRLZEZadWJFMVJNRXBZVjJ4b1MyVnRSbGhQV0ZaUlZrVldNVlJWVFRCa01IaHhVVmhPU2xKVk5IaFphMmhUVFZkT2RGWlViR2xpVmxsNFdrVm9TMkZIU2tSa01tUldVMFphY0ZsclpITmhiRTE1Vm1wV1YxSjZiSGxYYkdNd1QxZEtkVlp1VG1sUlZEQTVRMjFPTUdKcWIzZFBiRnBJVm01d2ExSnJiM2haYlRBeFlrZE9jRTVWU21GU01Gb3pXa1ZrVjJWWFRqVk9WbXhYVm5wV2QxcEZVazVrVmxwSVZtNXdhMU5GTVRGVlZFazFaRmRTU0ZKdVFtbGlWbG8xVkVkNFUySkhUWHBWYTFKcFVqQmFObGt6WXpsUVVYQndXa1J2ZDA5ck5WVlViWFJPWlcxT00xUnNaRVpsUm5CSVUxaGtVRlpGY0hSVWExSnJZVEE1UlZremFHRldSbXN4VjJ0U2EySlZOVVZVV0doT1VrWlZlRlF4WkU1a01EbEZWVmhrWVZZeFduQlhiWEJHWld4cmVWSlVTbEJTTUZweVZGZHdjbVZyTUhsVFdHeE9Va1pyZVZkdGNGSmtNRFZDVUZRd1BRcHBaRG93T2xsdFJYaGFSMUY0VFZkUk5FNTZUWHBPYWtVeldrUm5kMWxVV21oYWFrMTZUbFJLYkU1RVVYcFBWMDV0V2tSSk5GcHFhM3BPVkdSb1drUlpORnB0Um10T1ZFSnFXa2RWZVU5WFVtbE9WRlV6VGtSbk1GbG5QVDA9CmlkOjA6Tm1GbVpqWXpZVGd5T0RKaE9UVXlORE5pWXprNE9ETXpPV1U0WmpnMU5qa3hZVFJsTUdZM05EQTROakZoWldNMU5ERXlNMk5sWldJNU4yVTVNVFUzWXc9PQp0bWE6LTFbXTpjam8yT2pFS2RHdzZOam94Q213d09qWTZNUXBzWWpBNk5qb3dDbWt3T2pBNldrZEdNRmxSUFQwPQpleDoxMzpGYWxzZQp0bzo2OjA=
xUnit.net v3 In-Process Runner v0.1.1-pre.489+45aa94b38d (64-bit .NET 8.0.5)
  Starting:    TestRunner.Adapters.XUnit3.Tests.Container
    TheoryWithDisplayNameAndTypeArgument<String>(data: "data") [FAIL]
      System.InvalidOperationException : Late bound operations cannot be performed on types or methods for which ContainsGenericParameters is true.
      Stack Trace:
           at System.Reflection.RuntimeMethodInfo.ThrowNoInvokeException()
           at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
           at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
        --- End of stack trace from previous location ---
  Finished:    TestRunner.Adapters.XUnit3.Tests.Container
=== TEST EXECUTION SUMMARY ===
   TestRunner.Adapters.XUnit3.Tests.Container  Total: 1, Errors: 0, Failed: 1, Skipped: 0, Not Run: 0, Time: 0.062s

```

**Comment by bradwilson** [comment]
Can reproduce with 0.2.0-pre.8 (note that pre-enumerated theories are required to show the issue):

![image](https://github.com/xunit/xunit/assets/16944/cee96f19-ef96-4633-ad13-15cc0c8739db)

![image](https://github.com/xunit/xunit/assets/16944/8ad35e7d-082e-4cdb-8f64-c7b195406149)

**Comment by bradwilson** [comment]
Fixed in v3 `0.2.0-pre.18` https://xunit.net/docs/using-ci-builds
